packages/framework/store/src/transformer/job.ts

```diff
@@ -259,12 +267,12 @@ export class Job {
       SliceSnapshotSchema.parse(snapshot);
       const { content, pageVersion, workspaceVersion, workspaceId, pageId } =
         snapshot;
-      const contentBlocks: BlockModel[] = [];
-      for (const [i, block] of content.entries()) {
-        contentBlocks.push(
-          await this._batchSnapshotToBlock(block, doc, parent, (index ?? 0) + i)
-        );
-      }
+
+      const contentBlockPromises = content.map((block, i) =>
+        this._batchSnapshotToBlock(block, doc, parent, (index ?? 0) + i)
+      );
+      const contentBlocks = await Promise.all(contentBlockPromises);
```

Comment:
No Promise.all here.
It will cause race conditions where it is not executed in the order expected.

Reply:
As you can see you passed the initial test while the second test failed.

Reply:
Do you know what's the root cause of the race condition? Seems that `snapshotToBlock` also fails some test cases when running in parallel. If we can't enable `Promise.all`, the batch mechanism is not actually utilized.

Reply:
The reason is that doc.addblock needs a parent. If the parent is not there, everything could be wrong.